### PR TITLE
FIX: Fall back to sklearn for forest predictions on NaN data

### PR DESCRIPTION
**Pull request by david-cortes-intel**
## Description

ref https://github.com/uxlfoundation/scikit-learn-intelex/issues/3356

Solves the first point in the issue, in which passing data with missing values to prediction-related methods generates an error, by falling back to scikit-learn in those cases.

Note that this still doesn't produce correct results due to the second problem in the issue (incorrect assignment of missing branches).

---

<details>
<summary>Checklist:</summary>

**Completeness and readability**

- [x] Git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/uxlfoundation/scikit-learn-intelex/blob/main/doc/sources/contribute.rst) for details)_.
- [x] I have resolved any merge conflicts that might occur with the base branch.

**Testing**

- [x] I have run it locally and tested the changes extensively.
- [x] All CI jobs are green or I have provided justification why they aren't.
- [x] I have extended testing suite if new functionality was introduced in this PR.

</details>

### CODE REVIEW

**Review by ethanglaser — APPROVED** [review]
LGTM for now. Maybe in future we could come up with a creative solution to still allow non-nan rows to be inferenced in oneDAL.

File: sklearnex/ensemble/_forest.py [hunk 1]
```diff
@@ -354,6 +354,15 @@ def _onedal_cpu_supported(self, method_name, *data):
                     ),
                 ]
             )
+            if not dal_ready:
+                return patching_status
+            xp, _ = get_namespace(X)
+            has_nan = xp.any(xp.isnan(X))
```
Comment (icfaust) [inline sklearnex/ensemble/_forest.py]: unlike previous changes recently like this (adding nan checks) this is on the hot path. it is guaranteed to reduce performance. please move the finiteness check here and disable its use in validate_data in onedal_predict in order to prevent an additional load/eval.

**Review by icfaust — CHANGES_REQUESTED** [review]
sorry need to throw this on because of the approval

File: sklearnex/ensemble/_forest.py (hunk 1 above)
Comment (david-cortes-intel) [reply sklearnex/ensemble/_forest.py]: Fixed. But looks like the finite checker was also checking for infinites, which should be supported for predictions.

Comment (icfaust) [reply sklearnex/ensemble/_forest.py]: We have an optimized finite checker to improve perf, this unfortunately is still not as performant. What trick the inf+nan check does is that if it has an Inf, it will offload to sklearn and fail on sklearn-side (where performance loss isn't a big deal: I'd replicate similar behavior already in this file for fit, unless I am missing something).

Comment (david-cortes-intel) [reply sklearnex/ensemble/_forest.py]: Updated. Looks like scikit-learn indeed doesn't support infinites, even though oneDAL does.

**Review by icfaust — APPROVED** [review]
Just make sure to also modify the gpu version before merging. Thanks for doing the hard work.
